Utilities/File.cpp

```diff
@@ -161,6 +162,23 @@ static fs::error to_error(int e)
 
 #endif
 
+static std::string path_append(std::string path, std::string_view more)
+{
+	// TODO: Replace with starts_with and ends_with once C++20 is fully available
+	const bool src_slash = path.back() == '/';
+	const bool dst_slash = more.front() == '/';
+
+	if (!src_slash && !dst_slash)
+	{
+		path.push_back('/');
+	}
+	else if (src_slash && dst_slash)
+	{
+		more = more.substr(1);
+	}
+	return std::move(path.append(more));
```

Comment:
Isn't this redundant move which prevents copy elision?

Reply:
`path.append` returns `std::string&` which is `*this`, and when I stepped through it I noticed it was making a copy, not move.

Reply:
```suggestion
	path.append(more);
	return path;
```
?

Reply:
yep, copy elision does not work for function arguments.

Reply:
Indeed it's bit of a shame, but:

- `std::string` move is very lightweight
- This function is tiny, so inlining is likely

Reply:
just use const std::string& as argument, and copy it. dont return path.append directly but the object itself.

Reply:
Your other review comment kinda makes me want to change the way this function looks so this point might become outdated due to design change anyway.